### ISSUE TITLE
cannot run amplify codegen 

### ISSUE THREAD

**Issue by rodrigobastosv** [aws-amplify/amplify-codegen#137]
**Describe the bug**
I'm following the https://docs.amplify.aws/lib/datastore/getting-started/q/platform/flutter docs to implement DataStore on my flutter app. Doing Auth and Storage was just a breeze. But when trying to implement the DataStore i got an error on the codegen step 

After creating a new graphql api and try to run `amplify codegen models` i got this error:

```
amplify-codegen-appsync-model-plugin not support language target flutter. Supported codegen targets arr java, android, swift, ios, javascript, typescript
Error: amplify-codegen-appsync-model-plugin not support language target flutter. Supported codegen targets arr java, android, swift, ios, javascript, typescript
    at Object.buildGeneratesSection (/usr/lib/node_modules/@aws-amplify/cli/node_modules/amplify-codegen-appsync-model-plugin/src/preset.ts:214:15)
    at Object.generateModels (/usr/lib/node_modules/@aws-amplify/cli/node_modules/amplify-codegen/src/commands/models.js:44:67)
    at Object.run (/usr/lib/node_modules/@aws-amplify/cli/node_modules/amplify-codegen/commands/codegen/models.js:9:7)
    at Object.executeAmplifyCommand (/usr/lib/node_modules/@aws-amplify/cli/node_modules/amplify-codegen/src/amplify-plugin-index.js:9:3)
    at executePluginModuleCommand (/usr/lib/node_modules/@aws-amplify/cli/src/execution-manager.ts:161:3)
    at Object.executeCommand (/usr/lib/node_modules/@aws-amplify/cli/src/execution-manager.ts:25:5)
    at Object.run (/usr/lib/node_modules/@aws-amplify/cli/src/index.ts:92:5)
 ```

After some digging some tutorials advise to use the `amplify add codegen` command. But when i do i got the error:
```
flutter is not supported by codegen plugin
```

I'm a bit newbie with amplify and i'm feeling a bit lost.

**Platform**
Amplify Flutter current supports iOS and Android.  This issue is reproducible in (check all that apply):
[x] Android

**Comment by AaronZyLee** [comment]
Currently there are two kinds of codegen in amplify. One is the model generation for Datastore(high level), the other is the legacy type/doc generation for appsycn users(low level). Accordingly, the codegen commands differ when it comes to the two scenario above. 
Datastore codegen commands:
```
$ amplify codegen models
```
Appsync codegen commands:
```
$ amplify codegen
$ amplify codegen add
$ amplify codegen configure
$ amplify codegen remove
```

AFAIK flutter libraries only supports the first one(Datastore). The `amplify mock` is also for the second use cases. That's why if you have a flutter project and you run the commands intended for the second scenario will fail. We also do not have the flutter use cases in the doc of low-level codegen commands and mock api.

TL;DR, you can only use `amplify codegen models` in a flutter project and you don't need `amplify codegen/add` before running model-gen command.

**Comment by raywhiteside** [comment]
Thanks for the detailed background @AaronZyLee.

I am still a bit confused on where the line is between Amplify CLI and Amplify Flutter. The `amplify mock` guide [here](https://docs.amplify.aws/cli/usage/mock)  is under Amplify CLI, which is separate from the Amplify Libraries section that has options for Flutter/Android/iOS/Javascript. Furthermore it's not obvious from the `amplify mock` guide that any of the commands are specific to certain libraries. In fact the goal is to locally mock backend AWS resources like AppSync and S3 - so it's not clear why this has any dependency on the frontend library such as Amplify Flutter. I'm not saying there is no good reason for such a dependency, only that the dependency is neither documented nor intuitively obvious to me.

In my mind we are getting away from the original issue now and there are two new topics cropping up:

1. Can `amplify mock` and any other Amplify CLI flows that depend on low-level codegen commands be supported for Flutter? I'm not going to belabor this point, these flows are clearly useful and the request is parity.

2. Until or in place of 1. being resolved, can the documentation and error messages be improved? Besides your response on this thread, the situation here is not only non-functional but confusing. `flutter is not supported by codegen plugin` is definitely helpful, but being required and able to run `flutter codegen models` is very strange. Changing the error message to something like:

`flutter is only supported for "amplify codegen models", all other amplify codegen commands and flows that depend on them are not supported`

This would have saved me a lot of time.

**Comment by AaronZyLee** [comment]
@raywhiteside Thanks for point this out. 

For the first point, I contact the flutter team and they plan to support this feature on their roadmap, aka sending raw graphql request and `amplify mock` since this is a good workflow and requested feature by other developers. In addition, datastore does not use mock since it already has a local storage engine.

For the second, I agree that the error messaging is confusing and non-functional. I will fix this ASAP.
